### PR TITLE
fix(pattern): fixed circular dependency on card pattern

### PR DESCRIPTION
**Pull request by RaphaelAmadeu**
### Related Ticket(s)

#2179 

### Description

In order to fix a circular dependency issue between `CTA` and `Card`, I've replaced the usage of `CTA` for `LinkWithIcon` in the `Card` pattern.

### CODE REVIEW

File: packages/react/src/patterns/sub-patterns/Card/Card.js [hunk 1]
```diff
@@ -89,7 +89,9 @@ function renderFooter(cta, type) {
     cta && (
       <div className={`${prefix}--card__footer`}>
         {type !== 'link' ? (
-          <CTA style="text" {...cta} customClassName={`${prefix}--card__cta`} />
+          <LinkWithIcon href={cta.href}>
+            <span>{cta.copy}</span> <cta.icon.src />
+          </LinkWithIcon>
```
Comment (annawen1) [inline packages/react/src/patterns/sub-patterns/Card/Card.js:94]: I think what we also need to include here is the cta `type` (ie. local, external, jump) that gets lost when we replace with LinkWithIcon. We might need to build that functionality into LinkWithIcon and use the functions from `CTALogic` class (still waiting for that PR to be merged rn)

Comment (jeffchew) [reply packages/react/src/patterns/sub-patterns/Card/Card.js:94]: PR is merged in, I think it makes sense to do this approach.

Comment (RaphaelAmadeu) [reply packages/react/src/patterns/sub-patterns/Card/Card.js:111]: @jeffchew , @annawen1 
I've tried to implement the `CTALogic` methods on my latest commit

Comment (annawen1) [reply packages/react/src/patterns/sub-patterns/Card/Card.js:111]: hmm I was expecting the changes to be done on `LinkWithIcon` as well

Comment (RaphaelAmadeu) [reply packages/react/src/patterns/sub-patterns/Card/Card.js:106]: @annawen1 I've updated this PR, now placing the logic on `LinkWithIcon`, thanks!

**Comment by jeffchew** [comment]
@RaphaelAmadeu can you update your snapshot file?

File: packages/react/src/components/LinkWithIcon/LinkWithIcon.js [hunk 2]
```diff
@@ -20,14 +21,25 @@ const { prefix } = settings;
  * @param {object} props react proptypes
  * @param {object} props.children User content
  * @param {string} props.href Link url
+ * @param {string} props.type Link type
  * @returns {*} LinkWithIcon component
  */
-const LinkWithIcon = ({ children, href, ...props }) => {
+const LinkWithIcon = ({ children, href, type, ...props }) => {
```
Comment (jeffchew) [inline packages/react/src/components/LinkWithIcon/LinkWithIcon.js:27]: @RaphaelAmadeu @annawen1 I'm not sure if adding to here makes sense or not? If we do decide to do it this way, we would also need to set the icon by type as well which doesn't seem to be the case here. Perhaps we should only keep this on the `Card` level for now?

File: packages/react/src/patterns/sub-patterns/Card/Card.js [hunk 3]
```diff
@@ -85,15 +96,16 @@ function optionalContent(copy) {
  * @returns {object} JSX object
  */
 function renderFooter(cta, type) {
+  const Icon = CTALogic.iconSelector(cta.type);
   return (
     cta && (
       <div className={`${prefix}--card__footer`}>
         {type !== 'link' ? (
-          <CTA style="text" {...cta} customClassName={`${prefix}--card__cta`} />
+          <LinkWithIcon type={cta.type} href={cta.href}>
```
Comment (jeffchew) [inline packages/react/src/patterns/sub-patterns/Card/Card.js:104]: @RaphaelAmadeu see comment at https://github.com/carbon-design-system/ibm-dotcom-library/pull/2326/files#r425952465. If we're going to only keep this on the `Card` level, you can just pass in the `onClick` behavior here.

cc: @annawen1

File: packages/react/src/patterns/sub-patterns/Card/Card.js (hunk 1 above)
Comment (jeffchew) [reply packages/react/src/patterns/sub-patterns/Card/Card.js:106]: Sorry if the above wasn't too clear, I've made a couple of comments above but I'm wondering if we should keep `LinkWithIcon` as-is, but we can pass in the CTA logic on this level?

**Pull request merged**